### PR TITLE
unuse AVX512 build option add

### PR DESCRIPTION
**Pull request by aie2**
I want to build annoy without AVX512.
if env variable UNUSE_AVX512_ANNOY_COMPILE = YES, it builds without AVX512.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -32,6 +33,29 @@
 
 """
 
+def _get_AVX_macro_list():
+    cmd = """gcc -march=native -dM -E - < /dev/null | grep "__AVX512" | sort"""
```
Comment (erikbern) [inline setup.py:37]: what does this command do?

Comment (aie2) [reply setup.py:37]: The part of「gcc -march=native -dM -E - < /dev/null」
generates a list of ‘#define’ directives for all the macros defined during the execution of the preprocessor.
  c.f. https://gcc.gnu.org/onlinedocs/gcc/Preprocessor-Options.html#Preprocessor-Options

And, 「grep "__AVX512"」 extracts AVX512 macro.

Comment (erikbern) [reply setup.py:37]: this feels a bit hacky – i'm nervous this won't work on all platforms (windows definitely not)

Comment (aie2) [reply setup.py:37]: So, that's it. 
In windows gcc doesn't exists, so this code can't work.
sorry, I rethink.

**Comment by erikbern** [comment]
I'm hoping to add functionality to detect AVX capabilities in runtime at some point. I think that would be a much better way to disable it if the host process doesn't support it. So this might not be needed if I do that?

**Comment by aie2** [comment]
If you do that, I'm glad. Well then, I close this PR.

**Pull request closed without merging**